src/core/elf.h

```diff
@@ -3,43 +3,250 @@
 #define CORE_ELF_H
 
 #include <stdint.h>
+#include <stdlib.h>
 
-#define ELF_TYPE_EXECUTABLE 0x2
+#define ELF_TYPE_NONE               0 //No file type
+#define ELF_TYPE_REL                1 //Relocatable object file
+#define ELF_TYPE_EXECUTABLE         2 //Executable file
+#define ELF_TYPE_DYN                3 //Shared object file
+#define ELF_TYPE_CORE               4 //Core file
+#define ELF_TYPE_LOOS               0xFE00 //Environment-specific use
+#define ELF_TYPE_HIOS               0xFEFF //Environment-specific use
+#define ELF_TYPE_LOPROC             0xFF00 //Processor-specific use
+#define ELF_TYPE_HIPROC             0xFFFF //Processor-specific use
```

Comment:
Do we need all these types now?

Reply:
I thought it'd be a good idea to have all the details about the ELF64 spec. We could move them inside the .c instead, since they're not of much use otherwise.